File: frontend/src/types/product.ts
export type ProductFamily = 'Pergola' | 'Sunroom' | 'Railing' | 'Accessories';

export type ProductSpec = {
  label: string;
  value: string;
};

export type Product = {
  sku: string;
  name: string;
  family: ProductFamily;
  subType: string;
  application: string[];
  tier: string;
  featured: boolean;
  tagline: string;
  intro: string;
  image: string;
  imageFallback: string;
  specs: ProductSpec[];
  features: string[];
  accessories: string[];
  leadTime: string;
  moq: string;
  ctaLabel: string;
};

File: backend/schemas/product.py
from pydantic import BaseModel


class ProductSpec(BaseModel):
    label: str
    value: str


class Product(BaseModel):
    sku: str
    name: str
    family: str
    subType: str
    application: list[str]
    tier: str
    featured: bool = False
    tagline: str
    intro: str
    image: str
    imageFallback: str
    specs: list[ProductSpec]
    features: list[str]
    accessories: list[str] = []
    leadTime: str
    moq: str
    ctaLabel: str = "Inquire via WhatsApp"
